File: fm-domain/src/main/java/io/zeitwert/dddrive/persist/util/StringUtils.java
package io.zeitwert.dddrive.persist.util;

public class StringUtils {

	public static String toCamelCase(String name) {
		String[] parts = name.split("_");
		StringBuilder camelCaseString = new StringBuilder();
		for (String part : parts) {
			camelCaseString.append(StringUtils.toProperCase(part));
		}
		return camelCaseString.substring(0, 1).toLowerCase() + camelCaseString.substring(1);
	}

	public static String toProperCase(String name) {
		return name.substring(0, 1).toUpperCase() + name.substring(1).toLowerCase();
	}

	public static String toSnakeCase(String name) {
		return name.replaceAll("(.)(\\p{Upper})", "$1_$2").toLowerCase();
	}

}

File: fm-domain/src/main/java/io/zeitwert/dddrive/persist/util/StringUtils.kt
package io.zeitwert.dddrive.persist.util

import java.util.*

object StringUtils {

	fun toCamelCase(name: String): String {
		val parts = name.split("_".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
		val camelCaseString = StringBuilder()
		for (part in parts) {
			camelCaseString.append(toProperCase(part))
		}
		return camelCaseString.substring(0, 1).lowercase(Locale.getDefault()) + camelCaseString.substring(1)
	}

	fun toProperCase(name: String): String = name.substring(0, 1).uppercase(Locale.getDefault()) + name.substring(1).lowercase(Locale.getDefault())

	fun toSnakeCase(name: String): String = name.replace("(.)(\\p{Upper})".toRegex(), "$1_$2").lowercase(Locale.getDefault())

}
